documentation/Blob Structure.md

```diff
@@ -0,0 +1,106 @@
+
+- its own rollup transactions
+- any other rollup transactions it should process (eg. for embedded rollups)
+- any other data feeds it should process (eg. for synchrony or coordination)
+- any compression (or other data processing) algorithms it supports
+
+### Publication Hashes
+
+The rollup node should find relevant publication hashes in the `PublicationFeed` contract. This will include anything published by its inbox plus any other data feeds that it is watching. There may be multiple relevant publications that share the same blobs, and the rollup node will need to track which blobs it is processing for which reasons. In the simplest case it is just looking for its own rollup transactions inside publications made by its own inbox.
+
+### Publication Headers
+
+For each relevant hash, the publication is the concatenation of all the corresponding blobs. The rollup node should retrieve and validate the first 31-byte chunk, and interpret it as a header with the structure:
+
+- the version (set to zero) (1 byte)
+- header length in chunks, excluding this one (1 byte)
+    - I expect this to be zero most of the time
+- multiplier (1 byte)
+    - the log base 2 of a scaling factor for data offsets
+    - i.e. if this value is $x$ and an offset is specified as $y$, the relevant data starts at chunk $2^x \cdot y$
+- Up to 5 data type records consisting of
+    - data identifier (3 bytes)
+    - offset to corresponding segment after accounting for the multiplier (2 bytes)
+- Pad the remaining space with zeroes. Note that zero is an invalid data identifier, so there should be no confusion about what counts as padding.
+
+If the header is larger than one chunk, retrieve and validate the remaining chunks. They should all be structured with the same data type records (up to 6 in a chunk) and padded with zeros.
+
+Although we are not using binary search, I still think it is useful to have a deterministic structure. Therefore, in addition to validating the header structure, the rollup nodes should ensure:
+
+- all data identifiers are listed in ascending order
+- the actual data segments are also in the same order (i.e. the segment offsets are strictly increasing)
+- all segment offsets point within the publication
+    - note that the size of a segment is implicitly defined by where the next segment starts.
+
+Assuming the publication header is valid, the rollup node will know which chunks to retrieve and validate to find its relevant segments.
+
+
+### Compressed segments
+
+The data identifier will indicate which kind of compression (eg. gzip, lz4, etc) was used. After decompression, the segment should use the same structure so that it can be further subdivided. For simplicity:
+
+- the segment header should be outside the compression. This means the expected contents can be identified (and potentially ignored) before retrieving the whole segment and decompressing it.
```

Comment:
there is no definition of "segment header"...

Reply:
I included a comment emphasising that it's the same format as the publication header